docs/en/users/03_Main_view.md

```diff
@@ -239,7 +239,10 @@ For example, you can enter multiple instances of `f:`, `author:`, `intitle:`, `i
 Combining several search criteria implies a logical *and*, but the keyword ` OR `
 can be used to combine several search criteria with a logical *or* instead: `author:Dupont OR author:Dupond`
 
-When combining multiple negative operators (e.g. `!intitle:'thing1' AND !intitle:'thing2'`), you'll need to consider [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). See [this GitHub comment](https://github.com/FreshRSS/FreshRSS/issues/3236#issuecomment-891219460) for more details on the usage of `AND` vs `OR` when using combining multiple negative operators.
+When combining multiple negative operators (e.g. `!intitle:'thing1' AND !intitle:'thing2'`), you'll need
+to consider the order of logic. See [this GitHub comment](https://github.com/FreshRSS/FreshRSS/issues/3236#issuecomment-891219460)
```

Comment:
I'm not sure if that substantially addresses my remark given the problem we're trying to address. :-)

<hr>

When combining multiple negative operators, it's important to keep in mind that the individual conditions evaluate as either true or false, meaning there's a good chance you want to use `!intitle:'thing1' AND !intitle:'thing2'`. This is because when `!intitle:'thing1'` is true `!intitle:'thing2'` is probably false and vice versa.

Consider what you would do to reverse the example above, `author:Dupont OR author:Dupond`.

* Author: Dupont → matches Dupont, doesn't match Dupond ⇒ true
* Author: Dupond → doesn't match Dupont, matches Dupond ⇒ true
* Author: Dumas → doesn't match Dupont, doesn't match Dupond ⇒ false

If you simply add a `!`, `!author:Dupont OR !author:Dupond`, suddenly almost everything would be matched!

* Author: Dupont → matches !Dupond ⇒ true
* Author: Dupond → matches !Dupont ⇒ true
* Author: Dumas → matches !Dupont and matches !Dupond ⇒ true

While instead, if you use `AND`, as in `!author:Dupont AND !author:Dupond`

* Author: Dupont → matches !Dupond but not !Dupont ⇒ false
* Author: Dupond → matches !Dupont but not !Dupond ⇒ false
* Author: Dumas → matches !Dupont and !Dupond ⇒ true

Reply:
I think something along those lines is what we want here, but I'm still trying to figure out how to present it clearly.

Reply:
Ah, you were thinking about putting a lot more in-depth detail in the documentation. I figured that was unnecessary for a rather "edge" case.

Reply:
Well, if "consider the order of logic" would've been clear to you then fair enough but I don't think this phrase nor Wikipedia are very clear here. There's probably a better link?

Reply:
I found [the linked GitHub comment](https://github.com/FreshRSS/FreshRSS/issues/3236#issuecomment-891219460) from @math-GH informative enough, personally. (Admittedly, I still didn't *understand* it the first time I saw it.)

Reply:
I'd prefer to properly copy it over but alright, let's go with this then. But @Alkarex is that AND in there appropriate then?

Reply:
I'm fine with whatever, just explaining why I didn't bother being super detailed in this PR. 😅

Reply:
@Frenzie There is no support for a literal `AND` (which would be just some plain text), only `OR`. The logical `^` (AND) is implicit between all the parts of the query.
https://github.com/FreshRSS/FreshRSS/blob/edge/app/Models/BooleanSearch.php